### DISCUSSION TITLE
How to adjust bezier edge

### DISCUSSION THREAD

**Discussion by rileydanejohnston** [eclipse-glsp/glsp#809] [General] [answered]
Hello, I’m trying to implement a curved bezier line for edges but I seem to be missing something. I’m using the BezierCurveEdgeView and I changed the ‘routerKind’ to ‘bezier’ on the server side. I’m using node backend with the VSCode client. 

My questions are:
1. How can I control the edge path? The edge seems to be bound to a single fixed point. My goal is to have a gentle curve from point A to point B. I'd like the edge to adjust to where I move the elements, rather than the edge clinging to the fixed point (see photo).
2. I’d like the Bézier curve to have an arrow similar to the GEdgeView. Is it possible to do this with an argument or do I need to make a custom view?
3. Is there documentation where I can see the options/arguments to make changes to the bezier edge?
<img width="251" alt="Screenshot 2022-12-06 at 11 56 41 AM" src="https://user-images.githubusercontent.com/54558543/205986255-6d76363d-0523-48da-a179-1d1bec74f07c.png">

**Comment by planger** [comment, marked as answer]
Hi,

Bezier curves have been introduced in Sprotty (see [PR](https://github.com/eclipse/sprotty/pull/245)) and GLSP only "inherits" them from there. I'm afraid, I don't really have first-hand experience using them though. But there is an [example in Sprotty](https://github.com/kaisalmen/sprotty/blob/c15a155833431ff9b2c182e9a7746b5f191638d1/examples/classdiagram/src/model-source.ts#L466) which may help.

**Reply by rileydanejohnston** [reply]
Hi Philip,

Thank you for the response. The example you posted had part of the answer I was looking for. I needed to change my edge type to 'edge:bezier', which I did not know existed.

**Comment by rsoika** [comment]
@rileydanejohnston  in case you mean the anchor point where the edge starts from your model element, than GLSP and spotty provide already so called anchor computers for circles, rects and diamonds. If your model element is based on such a base shape no special implementation should be necessary

For example in [OpenBPMN](https://github.com/imixs/open-bpmn/) we use Manhattan routing and we just configure our model elements based on the default NodeViews:

```
configureModelElement(context, 'intermediateCatchEvent', EventNode, CircularNodeView);
configureModelElement(context, 'task', TaskNode, RoundedCornerNodeView);
configureModelElement(context, 'exclusiveGateway', GatewayNode, DiamondNodeView);
```

The router computes the anchor point automatically:

![image](https://user-images.githubusercontent.com/1339124/206045926-d2f1325b-6c71-4bae-a787-99dab3e1c4af.png)

If this is what you are asking for? In your picture it looks like your 'Test' label with a circle is just a test element?

**Reply by rileydanejohnston** [reply]
Hi Ralph, 

Thank you for the response. I ended up using the 'manhattan' routerKind, the BezierCurveEdgeView, and the 'edge:bezier' edge type. This is the result I was going for:

https://user-images.githubusercontent.com/54558543/206271269-30984617-9f8c-4159-b790-68be1685ba35.mov
